### ISSUE TITLE
Multiband Dynamic EQ

### ISSUE THREAD

**Issue by buerdman** [sadko4u/lsp-plugins#151]
I want to ask you what you think about a Multiband Dynamic EQ plugin. It works similar to a multiband compression plugin: when a signal crosses the threshold of a band the EQ becomes active. In contrast to a compressor, it allows you to intervene much more precisely. There is a Dynamic EQ of Zam plugins but only with one band and without an analyser. I think a multiband dynamic EQ with maybe 8 bands would fit very well into the LSP plugin collection.

A Multiband Sidechain Dynamic EQ with many bands more (maybe 32) would also be a great plugin. Similar to Trackspacer from Wavefactory (https://www.wavesfactory.com/audio-plugins/trackspacer/). What do you think?

**Comment by sadko4u** [comment]
I'm already planning to implement it but I'm currently very busy with migration to 1.2.0 release with decomposition of project tree into separate repositories.

Since dynamic equalizers are more computation-expensive than static ones, I believe we'll end up with 8 filters. But the final number of filters will be determined after we'll get some prototype working to measure the performance.

Probably I'll start the development at the end of this year or in the first quarter of the next year.

**Comment by buerdman** [comment]
Thats great to hear! I am very excited about it. I wish you all the best.

Am Mo., 19. Okt. 2020 um 13:12 Uhr schrieb Vladimir Sadovnikov <
notifications@github.com>:

**Comment by sadko4u** [comment]
@buerdman  I also want to get the more complete understanding of how the plugin should work.

According to what I'm currently thinking about the design, there should be:
- set of N filters which allow to set usual to parametric equalizer parameters: **type**, **slope**, **frequency**, **gain**, **Q**.
- each filter is additionally associated with some narrow frequency band, so we have **low freq** and **high freq** to specify the frequency range.
- when the signal for the associated frequency band crosses associated **threshold** value, the filter starts to decrease gain?
- are there any use cases when the gain should be increased?
- do we need to have more precise control over each filter? I mean, define a characteristic curve like for compressor/dynamic processor.

**Comment by buerdman** [comment]
That pretty much meets my expectations as well.

>     * each filter is additionally associated with some narrow frequency band, so we have **low freq** and **high freq** to specify the frequency range.

I'm not sure I understand you correctly. I would have thought that the frequency you can select for the band is also the frequency that analyzes the signal.

>     * when the signal for the associated frequency band crosses associated **threshold** value, the filter starts to decrease gain?
>     * are there any use cases when the gain should be increased?

I'm just thinking: If you can invert the threshold this could be a cool feature. I mean, if the signal gets too quiet you can amplify it.
Or if the plugin gets a sidechain input this can make sense for sound design. 

> 
>     * do we need to have more precise control over each filter? I mean, define a characteristic curve like for compressor/dynamic processor.

Maybe this is also a cool feature, but in my opinion not necessary.

**Comment by h1z1** [comment]
> are there any use cases when the gain should be increased?

Compensating for poorly encoded tracks.  ie mp3s and voice..   ffmpeg has mcompander fwiw, obviously without a gui.

**Comment by JamesPeters667** [comment]
Increasing a frequency level can be useful. It depends on your intent, just as your use of EQ can involve increasing frequencies (and not just decreasing them). There would be too many use cases for me to list. :)

Defining the characteristic curve of the filter would be welcome. Depending on this, the dynamic response of the EQ can be used for various effects.

**Comment by nullJaX** [comment]
I would really love to have this feature. I ended up many times using mutiband compressor to mimic this behavior, just using all bands in order to separate specific frequencies.

As for how it could work, I guess that it can be somewhat inspired on ie. Ozone 9 Dynamic EQ or FabFilter's one. So it would also be nice to include low/high pass and shelf curves.

Regarding the gain increase: for me it would behave like compressor in up mode for specific frequencies.

**Comment by x42** [comment]
> when the signal for the associated frequency band crosses associated threshold value, the filter starts to decrease gain?

When would this be useful? That sounds rather like a multiband compressor.

Dynamic EQ are usually two EQs with different settings. Which EQ is used depends on some condition. A common use-case are drum-hits: use EQ 1 for the onset then fade to EQ 2 for the tail.

**Comment by M5era** [comment]
Did any of you start to write a dynamic EQ? I am in the course of doing it myself and would love to exchange some ideas.

**Comment by sadko4u** [comment]
No, we're currently working on porting plugins to 1.2.0 with new core engine and graphical toolkit. Here are some previews:
![image](https://user-images.githubusercontent.com/12834464/135713080-6ee0b742-44a3-4af9-84a8-f16edd6509de.png)
![image](https://user-images.githubusercontent.com/12834464/135713127-042302e5-6634-4b99-b2af-0fbf5e714ab6.png)

After 1.2.0 release we'll continue delivering new plugins.

**Comment by magnetophon** [comment]
Wow, looks GOOD!

**Comment by talbym** [comment]
In my experience, this is currently the most advanced Dynamic EQ available for Linux, even FLOSS licensed:
https://github.com/ZL-Audio/ZLEqualizer
https://www.youtube.com/watch?v=VzmNqjWfE2Y
Maybe it is not necessary to redundantly invest time in creating such tool?

**Comment by magnetophon** [comment]
>  The distinguishing capability is pitch tracking. As the fundamental frequency moves in a melodic passage, the correction curve moves

That is also a cool plugin to have, but it's not what most people mean when they say dynamic EQ.
